### PR TITLE
feat: initial commit of GitLab in lead toolchain

### PR DESCRIPTION
**Pull request by jbankes**

### CODE REVIEW

File: modules/tools/gitlab/main.tf [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  source = "../../common/namespace"
+
+  resource_limit_cpu = "1"
+  resource_limit_memory = "4G"
+
+  namespace = "gitlab"
+}
+
+resource "helm_release" "gitlab" {
+  count = var.enable_gitlab ? 1 : 0
```
Comment (ChrisSchreiber) [inline modules/tools/gitlab/main.tf:11]: Let's move the feature flag from the resource to the parent module. This is a new feature with tf .13.

File: stages/apps/lead/gitlab.tf [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module "gitlab" {
```
Comment (ChrisSchreiber) [inline stages/apps/lead/gitlab.tf:1]: Move count (feature flag) here

File: modules/tools/gitlab/variables.tf [hunk 3]
```diff
@@ -0,0 +1,19 @@
+variable "enable_gitlab" {
```
Comment (ChrisSchreiber) [inline modules/tools/gitlab/variables.tf:1]: remove feature flag from the module itself

**Review by ChrisSchreiber — CHANGES_REQUESTED** [review]

File: modules/tools/gitlab/gitlab-values.tpl [hunk 4]
```diff
@@ -0,0 +1,22 @@
+global:
+  grafana:
+    enabled: false
+  hosts:
+    domain: ${gitlab_fqdn}
+  ingress:
+    class: toolchain-nginx
```
Comment (mrparkers) [inline modules/tools/gitlab/gitlab-values.tpl:7]: this should probably be a variable

File: modules/tools/gitlab/gitlab-values.tpl [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    annotations:
+      certmanager.k8s.io/issuer: letsencrypt-dns
```
Comment (mrparkers) [inline modules/tools/gitlab/gitlab-values.tpl:9]: and this

**Review by mrparkers — APPROVED** [review]

**Review by ChrisSchreiber — APPROVED** [review]

**Pull request merged**